### PR TITLE
Feature/bitfinex order book refactoring

### PR DESCRIPTION
**Pull request by GlebYarnykh**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -33,6 +33,18 @@
             <artifactId>jackson-core</artifactId>
             <version>2.9.3</version>
         </dependency>
+        <dependency>
+            <groupId>com.jsoniter</groupId>
+            <artifactId>jsoniter</artifactId>
+            <version>0.9.19</version>
+        </dependency>
+        <!-- https://mvnrepository.com/artifact/joda-time/joda-time -->
+        <dependency>
```
Comment (Mikhail-Borisov) [inline pom.xml]: Should not add that dependency

File: src/main/java/BaseObjects/LightWeightQuote.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    private final CurrencyPair<L, R> currencyPair;
+
+    private final int quoteId;
+
+    private final ExchangeRate<L, R> rate;
+
+    private BigDecimal leftAmount;
+
+    private MarketSide side;
+
```
Comment (Mikhail-Borisov) [inline src/main/java/BaseObjects/LightWeightQuote.java]: Probably do not need to have CurrencyPair field here.

File: src/main/java/BitfinexStreaming/BitfinexStreamingMarketDataService.java [hunk 3]
```diff
@@ -1,66 +1,89 @@
...
+    {
+        Observable<QuotesUpdate<L, R>> subscribedChannel = service.subscribeChannel(
+                BitfinexStreamingMarketDataService.orderbookChannelName,
+                new Object[]{currencyPair.toStringNoSlash(),
+                        defaultRawBookChannelPrecision,
+                        levels.getStringForSubscriptionMessage()})
+                .map(jsonQuotesUpdate ->
+                {
+                    if (BitfinexLevelUpdate.isSnapshot(jsonQuotesUpdate))
+                    {
```
Comment (Mikhail-Borisov) [inline src/main/java/BitfinexStreaming/BitfinexStreamingMarketDataService.java]: Should delete print statements

File: src/main/java/BitfinexStreaming/BitfinexStreamingMarketDataService.java [hunk 4]
```diff
@@ -1,66 +1,89 @@
...
+                        System.out.println("Parsing first snapshot");
+                        return new BitfinexLevelUpdate<>(currencyPair, jsonQuotesUpdate.get(1).iterator());
+                    }
```
Comment (Mikhail-Borisov) [inline src/main/java/BitfinexStreaming/BitfinexStreamingMarketDataService.java]: Delete this print too

File: src/main/java/OrderBook/BitfinexLevelUpdate.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+import BaseObjects.MarketSide;
+import Bitfinex.Bitfinex;
+import com.fasterxml.jackson.databind.JsonNode;
+import xchange.currency.Currency;
+import xchange.currency.CurrencyPair;
+
+import java.io.Serializable;
+import java.math.BigDecimal;
+import java.util.Iterator;
+
```
Comment (Mikhail-Borisov) [inline src/main/java/OrderBook/BitfinexLevelUpdate.java]: Name of the child is obscure

File: pom.xml (hunk 1 above)
Comment (GlebYarnykh) [reply pom.xml]: done

File: src/main/java/BaseObjects/LightWeightQuote.java (hunk 2 above)
Comment (GlebYarnykh) [reply src/main/java/BaseObjects/LightWeightQuote.java]: done

File: src/main/java/BitfinexStreaming/BitfinexStreamingMarketDataService.java (hunk 3 above)
Comment (GlebYarnykh) [reply src/main/java/BitfinexStreaming/BitfinexStreamingMarketDataService.java]: done

File: src/main/java/BitfinexStreaming/BitfinexStreamingMarketDataService.java (hunk 4 above)
Comment (GlebYarnykh) [reply src/main/java/BitfinexStreaming/BitfinexStreamingMarketDataService.java]: done

File: src/main/java/OrderBook/BitfinexLevelUpdate.java (hunk 5 above)
Comment (GlebYarnykh) [reply src/main/java/OrderBook/BitfinexLevelUpdate.java]: done

**Pull request merged**